src/logic/map_objects/pinned_note.cc

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2022 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "logic/map_objects/pinned_note.h"
+
+#include <memory>
+
+#include "graphic/rendertarget.h"
+#include "logic/game_data_error.h"
+#include "logic/player.h"
+#include "wui/interactive_player.h"
+
+namespace Widelands {
+
+class PinnedNoteDescr : public BobDescr {
+public:
+	PinnedNoteDescr(char const* const init_name, char const* const init_descname)
+	   : BobDescr(init_name,
+	              init_descname,
+	              MapObjectType::PINNED_NOTE,
+	              MapObjectDescr::OwnerType::kTribe) {
+	}
+
+	Bob& create_object() const override {
+		return *new PinnedNote();
+	}
+
+private:
+	DISALLOW_COPY_AND_ASSIGN(PinnedNoteDescr);
+};
+
+static const PinnedNoteDescr g_descr("pinned_note", "Pinned Note");
```

Comment:
Don't we need to make the Descname translatable her?

Reply:
This description should never be shown so no need. Same as for other singleton descriptions:

https://github.com/widelands/widelands/blob/51db95b540268e319d9508823c07f3de635ce0fc/src/economy/flag.cc#L44
https://github.com/widelands/widelands/blob/51db95b540268e319d9508823c07f3de635ce0fc/src/economy/portdock.cc#L42